subprojects/test-kit/src/main/java/org/gradle/testkit/runner/internal/ToolingApiGradleExecutor.java

```diff
@@ -167,6 +176,14 @@ public GradleExecutionResult run(GradleExecutionParameters parameters) {
         return new GradleExecutionResult(new BuildOperationParameters(targetGradleVersion, parameters.isEmbedded()), outputBuffer.readAsString(), tasks);
     }
 
+    private static void checkDeprecationWarning(GradleVersion targetGradleVersion) {
+        if (targetGradleVersion.compareTo(MINIMUM_SUPPORTED_GRADLE_VERSION) < 0) {
+            DeprecationLogger.deprecate("The version of Gradle you are using (%s) is deprecated with TestKit. TestKit will only support the last 5 major versions in future.")
```

Comment:
We could say something like:

```
TestKit uses the Tooling API to interact with Gradle builds, which no longer supports the version you are using (%s).
```

Then, instead of calling `undocumented`, you can link the tapi compatibility documentation: https://docs.gradle.org/current/userguide/third_party_integration.html#sec:embedding_compatibility